src/methods.jl

```diff
@@ -1,102 +1,147 @@
-    will be weighted by.
-- `bound_estimator::Function`: A function that takes in the function and the evaluation
-    point and returns a bound on the magnitude of the `length(grid)`th derivative.
+- `grid::SVector{P,Int}`: Multiples of the step size that the function will be evaluated at.
+- `coefs::SVector{P,Float64}`: Coefficients corresponding to the grid functions that the
+    function evaluations will be weighted by.
+- `coefs_neighbourhood::NTuple{3,SVector{P,Float64}}`: Sets of coefficients used for
+    estimating the magnitude of the derivative in a neighbourhood.
+- `condition::Float64`: Condition number. See See [`DEFAULT_CONDITION`](@ref).
+- `factor::Float64`: Factor number. See See [`DEFAULT_FACTOR`](@ref).
+- `max_range::Float64`: Maximum distance that a function is evaluated from the input at
+    which the derivative is estimated.
+- `∇f_magnitude_mult::Float64`: Internally computed quantity.
+- `f_error_mult::Float64`: Internally computed quantity.
+- `bound_estimator::FiniteDifferenceMethod`: A finite difference method that is tuned to
+    perform adaptation for this finite difference method.
 """
-struct FiniteDifferenceMethod{G<:AbstractVector, C<:AbstractVector, E<:Function}
-    grid::G
-    q::Int
-    coefs::C
+struct AdaptedFiniteDifferenceMethod{
+    P,
+    Q,
+    E<:FiniteDifferenceMethod
+} <: FiniteDifferenceMethod{P,Q}
+    grid::SVector{P,Int}
+    coefs::SVector{P,Float64}
+    coefs_neighbourhood::NTuple{3,SVector{P,Float64}}
+    condition::Float64
+    factor::Float64
+    max_range::Float64
+    ∇f_magnitude_mult::Float64
+    f_error_mult::Float64
     bound_estimator::E
 end
 
 """
     FiniteDifferenceMethod(
-        grid::AbstractVector,
+        grid::Vector{Int},
         q::Int;
-        condition::Real=DEFAULT_CONDITION
+        condition::Real=DEFAULT_CONDITION,
+        factor::Real=DEFAULT_FACTOR,
+        max_range::Real=Inf
     )
 
 Construct a finite difference method.
 
 # Arguments
-- `grid::Abstract`: The grid. See [`FiniteDifferenceMethod`](@ref).
+- `grid::Vector{Int}`: The grid. See [`AdaptedFiniteDifferenceMethod`](@ref) or
+    [`UnadaptedFiniteDifferenceMethod`](@ref).
 - `q::Int`: Order of the derivative to estimate.
+
+# Keywords
 - `condition::Real`: Condition number. See [`DEFAULT_CONDITION`](@ref).
+- `factor::Real`: Factor number. See [`DEFAULT_FACTOR`](@ref).
+- `max_range::Real=Inf`: Maximum distance that a function is evaluated from the input at
+    which the derivative is estimated.
 
 # Returns
 - `FiniteDifferenceMethod`: Specified finite difference method.
 """
 function FiniteDifferenceMethod(
-    grid::AbstractVector,
+    grid::Vector{Int},
     q::Int;
-    condition::Real=DEFAULT_CONDITION
+    condition::Real=DEFAULT_CONDITION,
+    factor::Real=DEFAULT_FACTOR,
+    max_range::Real=Inf
 )
     p = length(grid)
+    grid = SVector{p}(grid)  # Internally, we work with static vectors.
```

Comment:
Would it be better to provide two methods of `FiniteDifferenceMethod`, one of which accepts any old vector and converts it to an `SVector`? I can imagine a performance-conscious user might want to provide an `SVector` to start with.

Reply:
Yes, definitely. Added!